xdsl/irdl/attributes.py

```diff
@@ -353,7 +352,7 @@ def irdl_to_attr_constraint(
 
 @overload
 def irdl_to_attr_constraint(
-    irdl: Attribute | TypeVar | ConstraintVar,
+    irdl: Attribute | ConstraintVar,
     *,
     allow_type_var: bool = False,
     type_var_mapping: dict[TypeVar, AttrConstraint] | None = None,
```

Comment:
Are these overloads still needed since my earlier change?

Reply:
I'll remove the overload in the next PR, this one is the easy one. I'll open the next one as a stacked PR for @math-fehr to look at both before merging, seems like a delicate-enough part of the API.